sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/json/JacksonGenerator.scala

```diff
@@ -178,7 +178,7 @@ private[sql] class JacksonGenerator(
     var i = 0
     while (i < row.numFields) {
       val field = schema(i)
-      if (!row.isNullAt(i)) {
+      if (!row.isNullAt(i) || field.dataType == NullType) {
```

Comment:
I think this is a bit odd that we write out `null` since `null` has been ignored in JSON. For instance, when one field is all `null`s, it won't be able to infer the schema anyway. I think we should match the behaviour.

Maybe we should have an option to write all `null`s out like `dropFieldIfAllNull` but not sure how useful it would be.

Reply:
Also, I think CSV can be supported in the same way.

Reply:
It's trivial but removing `isReadPath` is also a trivial point. I think we should either provide an option to control it (which I think is a bit of an overkill) or keep the behaviour since other `null`s from other types are being ignored but writing out `null`s only for `NullType` sounds counter intuitive.

Reply:
Well, the motivation of is that in this PR I create **two API** for read/write. Later I find that read/write should support the same types. ORC doesn't support Null types, even if use provide a schema contains null column, the scan will still fail.
I can keep the current behavior of JSON data source. But I don't think we need the parameter `isReadPath`.

Reply:
Yea, reading and writing should be the same and that's ideal. Yup, let's keep the previous behaviour.

Ah, you mean ORC already doesn't support to read/write `NullType`? then ignore https://github.com/apache/spark/pull/23639#issuecomment-457427796. Yup, I agree with removing it then.